cmd/xl-v1-object.go

```diff
@@ -534,37 +543,41 @@ func (xl xlObjects) putObject(bucket string, object string, data *hash.Reader, m
 		// Check if an object is present as one of the parent dir.
 		// -- FIXME. (needs a new kind of lock).
 		// -- FIXME (this also causes performance issue when disks are down).
-		if xl.parentDirIsObject(bucket, path.Dir(object)) {
-			return ObjectInfo{}, toObjectErr(errors.Trace(errFileAccessDenied), bucket, object)
+		if xl.parentDirIsObject(ctx, bucket, path.Dir(object)) {
+			logger.LogIf(ctx, errFileAccessDenied)
+			return ObjectInfo{}, toObjectErr(errFileAccessDenied, bucket, object)
 		}
 
-		if err = xl.putObjectDir(minioMetaTmpBucket, tempObj, writeQuorum); err != nil {
-			return ObjectInfo{}, toObjectErr(errors.Trace(err), bucket, object)
+		if err = xl.putObjectDir(ctx, minioMetaTmpBucket, tempObj, writeQuorum); err != nil {
+			logger.LogIf(ctx, err)
```

Comment:
Possible duplicate logging.
`put ObjectDir` is supposed to take care or already takes care of the `err` error condition.

Reply:
done